x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/latest_vulnerabilities_group_renderer.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBadge,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+  EuiSkeletonTitle,
+  EuiText,
+  EuiTextBlockTruncate,
+  EuiToolTip,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import {
+  ECSField,
+  GroupPanelRenderer,
+  RawBucket,
+  StatRenderer,
+} from '@kbn/securitysolution-grouping/src';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { VulnerabilitiesGroupingAggregation } from './hooks/use_grouped_vulnerabilities';
+import { GROUPING_OPTIONS } from './constants';
+import { VULNERABILITIES_GROUPING_COUNTER } from './test_subjects';
+import { NULL_GROUPING_MESSAGES, NULL_GROUPING_UNIT, VULNERABILITIES } from './translations';
+import { getAbbreviatedNumber } from '../../common/utils/get_abbreviated_number';
+
+/**
+ * Return first non-null value. If the field contains an array, this will return the first value that isn't null. If the field isn't an array it'll be returned unless it's null.
+ */
+export function firstNonNullValue<T>(valueOrCollection: ECSField<T>): T | undefined {
```

Comment:
I think this exists for findings as well. Worth sharing the code between the two?

Reply:
and maybe it's true for some of the components in this file as well, the ones which are not specific for vulnerabilties

Reply:
That was a great idea, I ended up sharing the common components and utils in the cloud_security_grouping folder